src/app/components/new/new.component.ts

```diff
@@ -22,7 +30,35 @@ export class NewComponent {
     }
     return false;
   }
+
+  checkIfCommandDone(commandId: string): void {
+    this.commandService.isCommandDone(commandId)
+      .subscribe(response => {
+        this.isCommandDone = !!response;
+      });
+  }
+
+  checkAngularProject(): void {
+    this.commandService.isAngularProject()
+      .subscribe(response => this.isAngularProject = !!response);
+  }
+
   createNewProject() {
-    this.sendCommand.emit(this.options.createCommandString());
+    const requrst = new CommandRequest(this.options.createCommandString());
+    this.commandService.sendCommand(requrst)
+      .subscribe(commandId => {
+        console.log('started generating new project, ID:', commandId);
+        const timedStatusCheck = interval(1000)
+        .subscribe(x => {
```

Comment:
I'm interested what does this method returns.
by the way, if you're not using a return value you can do:

```typescript
  .subscribe(() => { 
```

Reply:
x represent the number of times the interval repeated